### PR TITLE
update HtmlPostProcessor for fal.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @hellosnow @ansyral @vicancy @superyyrrzz @qinezh

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

File: test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs [hunk 1]
```diff
@@ -12,21 +12,35 @@ namespace Microsoft.DocAsCode.Build.Engine.Tests
...
+        public ValidateBookmarkTest()
+        {
+            _outputFolder = GetRandomFolder();
+            EnvironmentContext.SetBaseDirectory(_outputFolder);
+            EnvironmentContext.SetOutputDirectory(_outputFolder);
+        }
+
+        public override void Dispose()
+        {
+            EnvironmentContext.Clean();
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs]: >EnvironmentContext.Clean(); [](start = 12, length = 27)

move to TestBase?

**Comment by vicancy** [comment]
:shipit:

File: test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs (hunk 1 above)
Comment (vwxyzh) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs]: We cannot move it to TestBase, unless all test base test is marked collection docfx sta.

**Pull request merged**